docs/concepts/canister-metadata.md

```diff
@@ -0,0 +1,121 @@
+# Canister Metadata
+
+## Overview
+
+Canisters can store custom metadata, which is available from the state tree at `/canister/<canister id>/metadata/<name>`.
+
+You can configure this metadata in dfx.json, per canister, in the `metadata` array.
+
+Here is a simple example:
+
+```json
+{
+  "canisters": {
+    "app_backend": {
+      "main": "src/app_backend/main.mo",
+      "type": "motoko"
+    },
+    "app_frontend": {
+      "dependencies": [
+        "app_backend"
+      ],
+      "frontend": {
+        "entrypoint": "src/app_frontend/src/index.html"
+      },
+      "source": [
+        "src/app_frontend/assets",
+        "dist/app_frontend/"
+      ],
+      "type": "assets",
+      "metadata": [
+        {
+          "name": "alternative-domains",
+          "visibility": "public",
+          "path": "src/app_frontend/metadata/alternative-domains.cbor"
+        }
+      ]
+    }
+  },
+  "version": 1
+}
+```
+## Fields
+
+The [JSON schema](../dfx-json-schema.json) also documents these fields.
```

Comment:
I'm not sure these documents will live next to each other in the docs. I would personally put dfx concepts [here](https://internetcomputer.org/docs/current/developer-docs/build/) and the json schema is [here](https://internetcomputer.org/docs/current/references/dfx-json-reference). This would break the link